### DISCUSSION TITLE
whisper.decocde() throws an "[W NNPACK.cpp:61] Could not initialize NNPACK! Reason: Unsupported hardware." error.

### DISCUSSION THREAD

**Discussion by smartic** [openai/whisper#2258] [Q&A] [answered]
Hello, 

I am starting with openai.whisper and, therefore, I have created a virtual machine in a vagrant environment - linux debian 12, (on a Windows host), and followed the installation process given in "https://github.com/openai/whisper ". Everything seems to start fine, both in the console and the first example with python3, but the second example that uses the statement "result = whisper.decode(model, mel, options)", throws the following error when executed:

#python3 whisper2.py
100% | █| 139M/139M [00:05<00:00, 27.3MiB/s]
Detected language: en
[W NNPACK.cpp:61] Could not initialize NNPACK! Reason: Unsupported hardware.
Traceback (most recent call last):
 File "/whisper/examples/whisper2.py", line 20, in <module>
 print(result["text"])
 ~~~~~~^^^^^^^^
TypeError: 'DecodingResult' object is not subscriptable

Does anyone know this problem and how it can be solved?
Thank you very much and best regards.

**Comment by glangford** [comment, marked as answer]
FYI in case the articles linked below are helpful (this is via Google search and I don't have direct experience with this error)

> `unsupported hardware` for nnpack typically happens if youre running on x86 cpu , but missing some common extended cpu instructions (either a old cpu or you're running inside a virtual machine that doesn't do full emulation properly - e.g., virtuabox is known for that).

Could not initialize NNPACK - StackOverflow
* https://stackoverflow.com/questions/69711410/could-not-initialize-nnpack

Running PyTorch Efficiently: Alternative Backends When NNPACK Fails
* https://python-code.dev/articles/411733062

**Comment by smartic** [comment]
Thank you very much for your information. It may be for the vagrant-virtualbox  environment.
